src/np_client.rs

```diff
@@ -65,50 +21,13 @@ pub struct NPClient {
     api_key: String,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct NPResponse {
-    pub success: bool,
-    pub data: Vec<Data>,
-    pub errors: Vec<String>,
-    pub warnings: Vec<serde_json::Value>,
-    pub info: serde_json::Value,
-    pub message_codes: Vec<String>,
-    pub error_codes: Vec<String>,
-    pub warning_codes: Vec<String>,
-    pub info_codes: Vec<serde_json::Value>,
-}
-
-#[derive(Debug, Deserialize, Serialize)]
-#[serde(rename_all_fields = "PascalCase")]
-#[serde(untagged)]
-pub enum Data {
-    Tracking(TrackingDoc),
-    Warehouse(Warehouse),
-    City(City),
-    Settlements {
-        total_count: u16,
-        addresses: Vec<Settlement>,
-    },
-    SettlementStreets {
-        total_count: u16,
-        addresses: Vec<StreetInfo>,
-    },
-}
-
-#[derive(Debug, Serialize)]
-pub enum Models {
-    TrackingDocument,
-    Address,
-}
-
 #[derive(Debug, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct NPRequest<'a> {
+pub struct NPRequest<'a, T: Serialize> {
     api_key: &'a str,
-    model_name: Models,
-    #[serde(flatten)]
-    called_method: Method<'a>,
+    model_name: &'a str,
+    called_method: &'a str,
```

Comment:
Maybe consider having enum with all used values instead of relying on hardcoded strings.